packages/react-router/src/Matches.tsx

```diff
@@ -603,10 +603,10 @@ export function useLoaderData<
     ...opts,
     select: (s) => {
       return typeof opts.select === 'function'
-        ? opts.select(s.loaderData)
+        ? opts.select(s.loaderData as TRouteMatch)
```

Comment:
shouldn't this be `TRouteMatch['loaderData']`?

Reply:
Yeah. This is weird. I will have to double check why this was complaining